### PR TITLE
210 docstrings

### PR DESCRIPTION
**Pull request by jarpri**

### CODE REVIEW

File: docs/conf.py
Comment (ninamiolane) [inline docs/conf.py:6]: No left over comments. If it's not needed, remove it.

File: topomodelx/base/message_passing.py
Comment (ninamiolane) [inline topomodelx/base/message_passing.py:11]: Verb in infinitiv form, according to numpy-style guidelines for docstrings.

**Review by ninamiolane — APPROVED** [review]
Looking good, thanks for this important contribution!

I've added some minor comments.

I see in the docstrings that two conventions are used.

Sometimes the rank is denoted k, as in: n_k_cells and "k-cells".
Sometimes it is denoted r.

For consistency and for a smooth user experience, we should only use one.

Let's use r.

Thus rename: n_k_cells to n_r_cells and k-cells to r-cells.

Thanks!

**Comment by ninamiolane** [comment]
@jarpri there is a github conflict with the can_train notebook. Github conflicts are not fun to solve on github. I'd recommend removing the can_train.ipynb from this PR and doing another PR for it. Your call though!

**Comment by papamarkou** [comment]
Thank you @jarpri. @ninamiolane, is this ready to be merged?

File: topomodelx/nn/hypergraph/allset.py
Comment (ffl096) [inline topomodelx/nn/hypergraph/allset.py:30]: Remove the `optional` here.

File: topomodelx/nn/cell/can_layer.py
Comment (ffl096) [inline topomodelx/nn/cell/can_layer.py:321]: Most of the time you use square brackets for the shape, here normal brackets. We should decide on one and be consistent with it.

File: topomodelx/nn/simplicial/sccn.py
Comment (ffl096) [inline topomodelx/nn/simplicial/sccn.py:52]: `dict` in lowercase, please (same below) 🙂

File: topomodelx/nn/simplicial/sccn_layer.py
Comment (ffl096) [inline topomodelx/nn/simplicial/sccn_layer.py:152]: `dict` in lowercase, please (same below) 🙂

File: topomodelx/nn/simplicial/sccnn.py
Comment (ffl096) [inline topomodelx/nn/simplicial/sccnn.py:217]:
```suggestion
        tensor, shape = [n_nodes, 2]
```

File: topomodelx/nn/simplicial/sccnn_layer.py
Comment (ffl096) [inline topomodelx/nn/simplicial/sccnn_layer.py:218]: Single return values do not have a name.

```suggestion
        torch.Tensor
```

File: topomodelx/nn/simplicial/sccnn_layer.py
Comment (ffl096) [inline topomodelx/nn/simplicial/sccnn_layer.py:275]:
```suggestion
        tuple, shape = (y_0,y_1,y_2)
```

File: topomodelx/nn/simplicial/scconv_layer.py
Comment (ffl096) [inline topomodelx/nn/simplicial/scconv_layer.py:154]:
```suggestion
        tuple of tensors, shape = (x0_out, x1_out, x2_out)
```

File: topomodelx/nn/simplicial/scnn_layer.py
Comment (ffl096) [inline topomodelx/nn/simplicial/scnn_layer.py:148]:
```suggestion
        torch.Tensor
```

**Review by ffl096 — CHANGES_REQUESTED** [review]
A few simple improvements, otherwise looks good to me!

File: topomodelx/nn/cell/can_layer.py
Comment (papamarkou) [reply topomodelx/nn/cell/can_layer.py:321]: Just tagging @jarpri so that it noted.

File: topomodelx/nn/hypergraph/allset.py
Comment (papamarkou) [reply topomodelx/nn/hypergraph/allset.py:30]: Tagging @jarpri.

File: topomodelx/nn/simplicial/sccn.py
Comment (papamarkou) [reply topomodelx/nn/simplicial/sccn.py:52]: Tagging @jarpri.

**Comment by papamarkou** [comment]
@ffl096, if it would be possible to squash and merge, it would be the most efficient; rebasing the branch might be quite painful right now for @jarpri, due to not having repeated the process.

**Comment by papamarkou** [comment]
@ffl096, @ninamiolane, we communicated with @jarpri, who kindly offered to try in the next couple of days to do the rebasing (as a learning exercise). I recommended that Jaro makes a backup of his work, so that if anything goes wrong, he can simply resubmit a fresh PR. Let's give Jaro the chance to give it a go, and we may coordinate.

**Comment by ffl096** [comment]
Don‘t worry, nothing is ever lost. If everything goes wrong, we can always find the orphan commit of the current state of this pr and restore it :)

**Comment by jarpri** [comment]
@ffl096, I've completed the rebase process. Could you please review the changes? I don't understand these conflicts. Thanks for your help and patience all!

**Comment by papamarkou** [comment]
@ffl096, @ninamiolane  I can see @jarpri completed the changes you suggested and has done the rebase as well. Is this PR ok to merge, or are there any real pending conflicts?

**Review by ffl096 — APPROVED** [review]
Looks good to me! I am on holidays until next Wednesday, so if somebody else can resolve the conflicts and merge :)

**Comment by papamarkou** [comment]
> Looks good to me! I am on holidays until next Wednesday, so if somebody else can resolve the conflicts and merge :)

Thank you @ffl096, I took a look at the conflicts. If I understand well, this is just the result of prining some docstrings only in the the notebooks. @ninamiolane, is this considered an issue, do we need to do anything about it, or is it ok?

**Comment by papamarkou** [comment]
@ffl096, just a gentle follow up if this could be merged?

**Comment by ffl096** [comment]
This is ready to be merged now. No need to squash commits

**Comment by papamarkou** [comment]
> This is ready to be merged now. No need to squash commits

Thank you very much @ffl096!

**Comment by ffl096** [comment]
(Just for clarification: I can't merge anymore since I am now the last pusher to this pull request. Someone else has to hit the button)

**Comment by papamarkou** [comment]
Ok, I did it :) Thank you so much for your help @ffl096!

**Pull request merged**
